libqtile/widget/mpris2widget.py

```diff
@@ -97,39 +130,45 @@ def update(self, interface_name, changed_properties, invalidated_properties):
 
         metadata = changed_properties.get('Metadata')
         if metadata:
+            metadata = metadata.value
             self.is_playing = True
-            self.displaytext = ' - '.join([
-                metadata.get(x)
-                if isinstance(metadata.get(x), dbus.String)
-                else ' + '.join(y for y in metadata.get(x) if isinstance(y, dbus.String))
-                for x in self.display_metadata if metadata.get(x)
-            ])
+
+            meta_list = []
+            for key in self.display_metadata:
+                val = metadata.get(key).value
+                if isinstance(val, str):
+                    meta_list.append(val)
+                elif isinstance(val, list):
+                    val = " - ".join((y for y in val if isinstance(y, str)))
+                    meta_list.append(val)
+
+            self.displaytext = ' - '.join(meta_list)
             self.displaytext.replace('\n', '')
 
         playbackstatus = changed_properties.get('PlaybackStatus')
-        if playbackstatus == 'Paused':
-            if self.stop_pause_text is not None:
-                self.is_playing = False
-                self.displaytext = self.stop_pause_text
-            elif self.displaytext:
-                self.is_playing = False
-                self.displaytext = 'Paused: {}'.format(self.displaytext)
+        if playbackstatus:
+            playbackstatus = playbackstatus.value
+            if playbackstatus == 'Paused':
```

Comment:
Can we make the diff simpler by not adding that extra level of indentation?

Reply:
Will add to the "to do" list.